Library/Homebrew/language/python.rb

```diff
@@ -228,9 +231,33 @@ def virtualenv_install_with_resources(using: nil, system_site_packages: true, wi
           python = T.must(wanted.first)
           python = "python3" if python == "python"
         end
+
+        venv_resources = if without.blank? && start_with.blank? && end_with.blank?
+          resources
+        else
+          remaining_resources = resources.to_h { |resource| [resource.name, resource] }
+          unless (unknown_resource = Array(without).find { |r| remaining_resources.delete(r).nil? }).nil?
+            raise ArgumentError, "#{__method__} `without` includes unknown resource \"#{unknown_resource}\""
+          end
+
+          start_with = Array(start_with)
+          start_with_resources = start_with.map { |r| remaining_resources.delete(r) }
+          unless (index = start_with_resources.find_index(nil)).nil?
+            raise ArgumentError, "#{__method__} `start_with` includes unknown resource \"#{start_with[index]}\""
+          end
+
+          end_with = Array(end_with)
+          end_with_resources = end_with.map { |r| remaining_resources.delete(r) }
+          unless (index = end_with_resources.find_index(nil)).nil?
+            raise ArgumentError, "#{__method__} `end_with` includes unknown resource \"#{end_with[index]}\""
+          end
```

Comment:
```suggestion
          end_with_resources = Array(end_with).map do |r|
            remaining_resources.delete(r) ||
              raise(ArgumentError, "#{__method__} `end_with` includes unknown resource \"#{r}\"")
          end
```

Nit: Here's an alternative syntax. I'm meh on both the current logic and my alternative. Just an idea. This same pattern could be used for without, start with and end with.

Note: I have not tested this but it should work.